docs/specs/code/TRANSACTION_MEMO.md

```diff
@@ -0,0 +1,239 @@
+
+ * The file will eventually get bigger and bigger as memos are added. A file versioning scheme could be considered in that case.
+ * Querying for a subset of memos (ie. last 10 memos) would require querying for the entire file from the external service.
+ * If two instances of the same wallet upload a memo at the same time, there will be a merge conflict. In the case of Dropbox, they handle this scenario by creating a **conflicting copy** of the file and maintining both files, passing on the responsability of merging them to the app.
+
+### Single folder per wallet, single file per transaction.
+
+The second strategy is to store memos in separated files inside a single folder per wallet. There would be one file per transaction, in which the content of the file is the corresponding UTF-8 enconded memo. The advantages of this approach are the following:
+
+ * Merge conflicts are avoided because two instances of the same wallet will not conflict if they write different files to the same folder. Two instances of the same wallet will never create the same transaction, therefore it's safe to assume that they will write different files.
+ * It would be easier to query for a subset of memos because in this case it implies querying for specific files. If the file doesn't exists, then the transaction doesn't have a memo. 
+
+The main disadvantage of this approach would be to handle a number of files instead of one. That is, encrypting, signing, uploading and downloading multiple files.
+
+Despite of this minor setback, we consider this strategy to be the most suitable for implementing the memo feature.
+
+## Privacy
+
+A memo could include sensitive or personal information about either the sender, the receiver or the transaction itself. In order to maintain privacy, each file containing a memo must be locally encrypted before uploaded to the external service. In the case of an unauthorized access to the external service, memo data will be unreadable. The user must acknowledge, however, that connecting to an external service will be at his/her own risk. If the user loses access to his/her account in the external service, the files could expose metadata about transactions history based on file creation dates.
+
+## Encryption
+
+To safely store data in the external service we have to use encryption and digital signatures. However, before going any further we must notice an important limitation about **hardware wallets**. For security reasons, hardware wallets only provide access to the device's public keys, **NOT** private keys. To address this, we discuss two approaches:
+
+### Public key encryption & password encryption
+
+This approach relies on using solely the *account* public key (available in all wallets) as input to create a new private key chain. More specifically:
+
+* Take the **Public Deriver** public key (*aka* account public key) and hash it with `Blake2b`.
```

Comment:
As I mentioned in Slack, the Public Deriver is not guaranteed to actually contain a public key. 
This is useful  for users  who want a “balance tracker” where you input a set of addresses and Yoroi monitors the balance for those addresses for you

For this case we would need to ask the user to input a memo password when they create their wallet and hash that as the entropy and then store that locally

Reply:
We will continue the discussion about this in the key derivation spec: https://github.com/Emurgo/yoroi-frontend/pull/899